src/app/task/conversation/conversation.component.ts

```diff
@@ -20,49 +22,68 @@ export class ConversationComponent implements OnInit {
   task: Task;
   public userTask: UserTask;
   public messages: Message[];
+  public comments: Comment[];
   public notExistingMessage: string;
   public userMessage: string;
   public recentMessages: Message[] = [];
   public userId: number;
   public minValueLength = 2;
+  public userTaskId: number;
 
   constructor(public dialogRef: MatDialogRef<ConversationComponent>,
-    private  alertwindow: AlertWindowsComponent,
+    private alertwindow: AlertWindowsComponent,
     private taskService: TaskService,
     private authService: AuthService,
+    private commentService: CommentService,
     private httpStatusCodeService: HttpStatusCodeService,
     @Inject(MAT_DIALOG_DATA) public data: any) {
-    this.task = data.task;
-    this.userTask = data.userTask;
+    this.task = data.task || {};
+    this.userTask = data.userTask || {};
+    this.userTaskId = data.task.Id || {};
   }
 
   notExistingUserTask() {
     this.dialogRef.close();
-    this.alertwindow.openSnackBar('You are not asigned to this plan!' , 'Ok');
+    this.alertwindow.openSnackBar('You are not asigned to this plan!', 'Ok');
   }
 
-  getUTMessages(userTaskId: number) {
-    this.taskService.getMessages(userTaskId).subscribe(
-      mes => {
-        if (mes.body.length === 0) {
-          this.notExistingMessage = 'Your conversation with mentor is empty. \n' +
-            'Ask some questions, if you have any.';
-        } else {
-          this.messages = mes.body;
-        }
-      });
+  async getUTMessages(userTaskId: number) {
+    if (this.authService.isMentor()) {
+      const _comment: any = {
+        Id: 0,
+        Text: 'string',
+        CreatorId: this.authService.getUserId(),
+        CreatorFullName: this.authService.getUserFullName(),
+        CreateDate: new Date().toISOString(),
+        ModDate: new Date().toISOString(),
+      };
+
+      // const comments = await this.commentService.addComment(_comment, userTaskId).toPromise(); // (i => console.log(i));
+      // console.log(comments);
+      // this.comments = comments || [];
+    } else {
+      this.taskService.getMessages(userTaskId).subscribe(
+        mes => {
+          if ((mes.body || []).length === 0) {
```

Comment:
instead of creating empty array check if `mes.body` is an array and then check its `length`, [see here](https://stackoverflow.com/a/24403771/2974754).